x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/RestPostConnectorSyncJobAction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.syncjob.action;
+
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.application.EnterpriseSearch;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+
+public class RestPostConnectorSyncJobAction extends BaseRestHandler {
+
+    @Override
+    public String getName() {
+        return "connector_sync_job_post_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/" + EnterpriseSearch.CONNECTOR_SYNC_JOB_API_ENDPOINT));
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        PostConnectorSyncJobAction.Request request = PostConnectorSyncJobAction.Request.fromXContentBytes(
+            restRequest.content(),
+            restRequest.getXContentType()
+        );
+
+        return channel -> client.execute(
+            PostConnectorSyncJobAction.INSTANCE,
+            request,
+            new RestToXContentListener<>(channel, r -> RestStatus.CREATED, r -> null)
```

Comment:
Will this always be a 201, or can it sometimes be a 200?

Reply:
I think this returns either `201` or it fails with some `4XX` status as this endpoint is only used for creating and not updating sync jobs. Citing this [StackOverflow post](https://stackoverflow.com/a/44396004) and this [rfc](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.2) it seems to me that `201` indicating that something was created + returning a reference to the created object (in this case `id`) seems to favour `201` over `200`, when the request was successful.

WDYT?

Reply:
I'm good with that.